lib/cocoapods/validator.rb

```diff
@@ -55,6 +74,28 @@ def file
       @linter.file
     end
 
+    # Returns a list of platforms to lint for a given Specification
+    #
+    # @param [Specification] spec
+    #         The specification to lint
+    #
+    # @return [Array<Platform>] platforms to lint for the given specification
+    #
+    def platforms_to_lint(spec)
+      return spec.available_platforms if @platforms.empty?
+
+      # Validate that the platforms specified are actually supported by the spec
+      results = @platforms.map do |platform|
+        matching_platform = spec.available_platforms.find { |p| p.name == platform.name }
```

Comment:
Need to do this so we keep the `target` value, since the platforms in `VALID_PLATFORMS` only have names associated with them